test/web.js

```diff
@@ -10,710 +10,801 @@ var testCreds = require("./testCredentials");
 var url = require("url");
 
 lab.experiment("OAuth", function() {
-  var s = server({
-    loginAPI: "http://localhost:3232",
-    cookieSecret: "test",
-    oauth_clients: testCreds.clients,
-    authCodes: testCreds.authCodes,
-    accessTokens: testCreds.accessTokens,
-    enableCSRF: false
-  });
+  var s, s2, ls;
+  const cookieSecret = "test".padEnd(32, "test");
 
-  var s2 = server({
-    loginAPI: "http://localhost:3232",
-    cookieSecret: "test",
-    oauth_clients: testCreds.clients,
-    authCodes: testCreds.authCodes,
-    accessTokens: testCreds.accessTokens,
-    enableCSRF: true
-  });
+  lab.before(async () => {
+    s = await server({
+      loginAPI: "http://localhost:3232",
+      cookieSecret,
+      oauth_clients: testCreds.clients,
+      authCodes: testCreds.authCodes,
+      accessTokens: testCreds.accessTokens,
+      enableCSRF: false
+    });
+
+    s2 = await server({
+      loginAPI: "http://localhost:3232",
+      cookieSecret,
+      oauth_clients: testCreds.clients,
+      authCodes: testCreds.authCodes,
+      accessTokens: testCreds.accessTokens,
+      enableCSRF: true
+    });
 
-  var ls = loginServer({
-    loginAPI: "http://localhost:3232"
+    ls = loginServer({
+      loginAPI: "http://localhost:3232"
+    });
   });
 
-  lab.test("GET / Redirects to /signup", function(done) {
+  lab.test("GET / Redirects to /signup", async () => {
```

Comment:
you can skip reviewing most of the tests except for the ones where I comment on about a real change that I made.